### PR TITLE
Using psql command instead of postgres in initialScript

### PR DESCRIPTION
**Pull request by rsrohitsingh682**
Resolves : https://github.com/juspay/services-flake/issues/63

Key changes in this PR:
- Use `psql` instead of `postgres`, as a result use `pg_ctl` to temporarily start the postgres server in `${name}-init` process.
- Avoid creating a custom `postgresPkg` with extensions by using [apply](https://github.com/juspay/services-flake/pull/66/commits/6940b2ca4986ae73c29d7f69f25fa3b4e5b894c2) attribute provided by `mkOption`
- Refactor setup script, use `writeShellApplication` and resolve shell-check errors

### CODE REVIEW

**Comment by shivaraj-bh** [comment]
The CI is going into an infinite loop while running the `postgres_test.nix`, stopping it for now.

File: example/flake.nix [hunk 1]
```diff
@@ -51,7 +51,7 @@
               {
                 environment.PGWEB_DATABASE_URL = "postgres://$USER@${pgcfg.listen_addresses}:${builtins.toString pgcfg.port}/${dbName}";
                 command = pkgs.pgweb;
-                depends_on."pg1".condition = "process_healthy";
+                depends_on."pg1-init".condition = "process_completed_successfully";
```
Comment (srid) [inline example/flake.nix:54]: Why this change?

Comment (shivaraj-bh) [reply example/flake.nix:54]: Earlier we would use `postgres` CLI command to load the `initialScript` but as explained in https://github.com/juspay/services-flake/issues/63, it might not be the ideal way to load dumps/scripts. Switching to `psql` to do the same would require the `pg1-init` to run after `pg1`, hence the dependency changes. Although the naming of `pg1-init` should be changed as it is misleading.

Comment (shivaraj-bh) [reply example/flake.nix:54]: This would prove to be useful while resolving https://github.com/juspay/services-flake/issues/64 as `psql` works well with most of the dumps, but the `postgres` CLI command doesn't as seen in our case while trying to load these dumps: https://github.com/nammayatri/nammayatri/blob/94d8212c20043ad179e5b51b599deab4af8a4e8b/Backend/nix/arion-configuration.nix#L23-L38

Comment (shivaraj-bh) [reply example/flake.nix]: @srid I have added comments to move back to depending on `pg1` as we can use `pg_ctl` to temporarily start the postgres server while `pg1-init` is running.

Comment (shivaraj-bh) [reply example/flake.nix]: I have also refactored the service a bit, a lot of it can still be split into parts but this seemed enough based on what the PR is titled.

Comment (srid) [reply example/flake.nix]: LGTM

**Review by shivaraj-bh — APPROVED** [review]

**Comment by shivaraj-bh** [comment]
@rsrohitsingh682 will merge this today if there is nothing more to add
